### ISSUE TITLE
https://github.com/enovance/puppet-openstack-cloud/pull/757 results in the whole galera cluster going down during upgrades and not restarting

### ISSUE THREAD

**Issue by tremble** [enovance/puppet-openstack-cloud#759]
Background : We're testing upgrades from RH7-I.1.2 to RH7-I.1.3 and 757 got pulled into our I.1.3 branch to avoid a split brained cluster (which we encountered during testing)

During the upgrade process all but one mysql node is shutdown, with 1 then started back up (with --wsrep-new-cluster) by the ansible scripts.

Now, once the Puppet run kicks off we're in a race condition situation, especially since order isn't guaranteed.

If the "master" node updates the bootstrap script before mysql gets restarted on the other nodes then the service restart which is triggered can leave you with a mysql cluster with no nodes up.  Since --wsrep-new-cluster has now been removed from the init script, mysql now gets stuck trying to restart.

**Comment by EmilienM** [comment]
Have you tried to
* stop mysql-bootstrap service on the master that should trigger a master change in the cluster
* start mysql regular process (mariadb) on the master
* run the upgrade

I suspect mysql-bootstrap tries to start anywhere else but master node.

**Comment by tremble** [comment]
The ansible scripts currently don't *try* to bring any of the other mysql nodes back up.  This then leaves you  the situation after the "before_config" ansible scripts have been run where the cluster has been "bootstrapped" (and the fact picks up on this), but only one node is up.

Since mysql-bootstrap no longer actually bootstraps the cluster, if this gets restarted by puppet before the other nodes have been started then your cluster is totally shut down and won't start without manual intervention.

I can bring the service back up manually, but this means that the "upgrade" jenkins job can't be run without intervention.  I've got a patched version of the ansible scripts which I'm trying out to work around this.

**Comment by tremble** [comment]
https://github.com/enovance/edeploy-roles/compare/master...tremble:cloud_puppet_759 is what I'm trying at the minute

**Comment by EmilienM** [comment]
@tremble sorry for that, I think they backported something that made fail the upgrade.

**Comment by tremble** [comment]
Yeah, I suspect it'll break things for future MySQL upgrades too.
